### PR TITLE
validate content size a la ssb-validate

### PR DESCRIPTION
**Pull request by mixmix**
# :fire: :bug: 

this module has been letting messages through that other peers consider invalid.
The reference implementation put a cap on the content size, see https://github.com/ssbc/ssb-validate/blob/main/index.js#L111

It appears that this module has not honoured that.
This means that there will likely be messages in the wild which are invalid for some peers and valid for others.

I think we need to consider how to handle that given it might have far reaching + feed-killing effects

### CODE REVIEW

**Comment by mixmix** [comment]
Proposal - think we might want to change out validation to what planetary is doing:
- don't allow writing > 8kb
- allow reading any size

**Comment by staltz** [comment]
Thanks, well spotted. It seems that this validation was on my mind when I was writing this module, but for some reason I forgot it.

> It appears that this module has not honoured that.

Gladly this module is not yet in production in Manyverse (it will be put this month into production!) so your PR came at the perfect time. I don't believe ssb-classic is used anywhere in production, maybe in Perihelion. Not sure.

The style changes were driving me crazy in this PR, but I realized that that's because we didn't have .prettierrc (we should) and that format-code is run automatically for every commit. I now added the prettierrc config and it should revert some of the code style changes.

File: validation.js [hunk 1]
```diff
@@ -142,24 +143,30 @@ function validateHash(msgVal) {
...
+  if (typeof content === 'string') {
+    if (!content.endsWith('.box') && !content.endsWith('.box2')) {
+      return new Error(
+        'invalid message: string content must end with .box or .box2, on feed: ' +
+          msgVal.author
+      )
+    }
+    if (content.length > SIZE_LIMIT) {
+      return new Error(
+        'invalid message: string content must be less than 8192 characters. Current size is ' +
```
Comment (staltz) [inline validation.js]:
```suggestion
        'invalid message: string content must be at most 8192 characters. Current size is ' +
```

File: validation.js [hunk 2]
```diff
@@ -174,102 +181,109 @@ function validateContent(msgVal) {
       // prettier-ignore
       return new Error('invalid message: content type must be longer than 2 characters, on feed: ' + msgVal.author);
     }
+    const size = JSON.stringify(content, null, 2).length
+    if (size > SIZE_LIMIT) {
+      return new Error(
+        'invalid message: encoded content must be less than 8192 characters. Current size is ' +
```
Comment (staltz) [inline validation.js]:
```suggestion
        'invalid message: encoded content must be at most 8192 characters. Current size is ' +
```

**Review by staltz — CHANGES_REQUESTED** [review]

**Comment by mixmix** [comment]
Please take over and merge if it's patches

**Comment by staltz** [comment]
@mixmix There's still a test that needs to be moved from https://github.com/ssbc/ssb-db2/pull/394 to here. I won't "take over" this PR because it wasn't my initiative. I had other plans for my day.

**Comment by mixmix** [comment]
@staltz added tests

**Comment by staltz** [comment]
Thanks!

**Pull request merged**